### PR TITLE
task(upgrade): write upgrade guide and bash script

### PR DESCRIPTION
**Pull request by rich-bugsnag**
## Goal

write upgrade guide and bash script

## Changeset

- Updated Upgrading.md
- Added a bash script that removes V5 from unity automatically

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -1,6 +1,131 @@
...
+
+#### Deprecations
+
+The following properties/methods have been removed from the `Bugsnag` client:
+
+| v5.x API                                                           | v6.x API                                                          |
+| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
+| `Bugsnag.LeaveBreadcrumb(message)`                                 | Replaced by `Bugsnag.LeaveBreadcrumb(message, metadata, type)`    |
+| `Bugsnag.LeaveBreadcrumb(message, type, metadata)`                 | Replaced by `Bugsnag.LeaveBreadcrumb(message, metadata, type)`    |
+| `Bugsnag.LeaveBreadcrumb(breadcrumb)`                              | Replaced by `Bugsnag.LeaveBreadcrumb(message, metadata, type)`    |
```
Comment (tomlongridge) [inline UPGRADING.md:90]: The second and third arguments are optional, so I don't think it's true to say that `LeaveBreadcrumb(message)` is deprecated?

File: UPGRADING.md [hunk 2]
```diff
@@ -1,6 +1,131 @@
...
+
+### Callbacks
+
+#### Additions
+
+The following methods have been added to the `Bugsnag` class: 
+
+| Property/Method                                                    | Usage                                                             |
+| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
+| `AddOnError` | Add a callback to modify or discard a event before it is recorded.  LINK_DOCS   |
```
Comment (tomlongridge) [inline UPGRADING.md:101]:
```suggestion
| `AddOnError` | Add a callback to modify or discard an event before it is captured (Android JVM exceptions, ANRs and handled events only).  LINK_DOCS   | 
```

File: UPGRADING.md [hunk 3]
```diff
@@ -1,6 +1,131 @@
...
+### Callbacks
+
+#### Additions
+
+The following methods have been added to the `Bugsnag` class: 
+
+| Property/Method                                                    | Usage                                                             |
+| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
+| `AddOnError` | Add a callback to modify or discard a event before it is recorded.  LINK_DOCS   | 
+|  `AddOnSendError`  | Add a callback to modify or discard a event before it is sent (including native Cocoa and Android events).  LINK_DOCS   |
```
Comment (tomlongridge) [inline UPGRADING.md:102]:
```suggestion
| `AddOnSendError`  | Add a callback to modify or discard any event before it is sent (including all iOS/macOS and native Android crashes).  LINK_DOCS   | 
```

File: UPGRADING.md [hunk 4]
```diff
@@ -1,6 +1,131 @@
...
+
+#### Deprecations
+
+The following methods have been removed from the `Bugsnag` client:
+
+| v5.x API                                                           | v6.x API                                                          |
+| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
+| `Bugsnag.SetAutoNotify(bool)`                                      | Deprecated - no longer public API.  Callbacks should now be used to stop sending particular errors at runtime|
+| `Bugsnag.SetAutoDetectErrors(bool)`                                      | Deprecated - no longer public API.  Callbacks should now be used to stop sending particular errors at runtime|
+| `Bugsnag.SetAutoDetectAnrs(bool)`                                  | Deprecated - no longer public API.  Callbacks should now be used to stop sending particular errors at runtime|
```
Comment (tomlongridge) [inline UPGRADING.md:124]: We should say these are replaced by the `AutoDetectErrors` and `EnabledErrorTypes` configuration options or controlled in an on-send callback.

**Review by tomlongridge — COMMENTED** [review]
A first pass - will need to re-read again, maybe once all docs changes are done.

Are there also lots of configuration options being deprecated that in v5 are just marked with a warning? e.g. `Endpoint`

File: UPGRADING.md (hunk 1 above)
Comment (rich-bugsnag) [reply UPGRADING.md]: Fixed 👍

File: UPGRADING.md (hunk 4 above)
Comment (rich-bugsnag) [reply UPGRADING.md]: Done 👍

**Review by tomlongridge — APPROVED** [review]

**Pull request merged**